src/planwise/client/scenarios/handlers.cljs

```diff
@@ -142,20 +142,27 @@
                 (assoc-in [:current-scenario :name] name)
                 (assoc-in [:view-state] :current-scenario))})))
 
+(defn new-provider-name
+  [changeset]
+  (let [new-providers (filter #(= (:action %) "create-provider") changeset)]
+    (if (empty? new-providers)
+      "New provider 0"
+      (let [vals (mapv (fn [p] (->> (:name p) (re-find #"\d+") js/parseInt)) new-providers)]
```

Comment:
This is a nice piece of code! But, it "fails" if the name doesn't contain digits (which shouldn't happen, I know), since `(js/parseInt nil)` is NaN. You can fix replacing `js/parseInt` by `int` (which in Clojurescript does what you want, but not in Clojure) or `(fnil js/parseInt 0)`. Ie:
```clojure
(mapv (fn [p] (->> (:name p) (re-find #"\d+") int)) new-providers)
```
or
```clojure
(mapv (fn [p] (->> (:name p) (re-find #"\d+") ((fnil js/parseInt 0)))) new-providers)
```